File: 645-set-mismatch/set-mismatch.java
class Solution {
    public int[] findErrorNums(int[] nums) {
        int[]arr=new int[2];
        Arrays.sort(nums);
        int n=nums.length;
        int r_s=n*(n+1)/2;
        int a_s=0;
        for(int i=1;i<nums.length;i++){
            if(nums[i-1]==nums[i]){
                arr[0]=nums[i];
            }
        }
        for(int num:nums){
            a_s+=num;
        }
        arr[1]=r_s-(a_s-arr[0]);
        
       
        return arr;
        
    }
}

File: 645-set-mismatch/set-mismatch.cpp
class Solution {
public:
    vector<int> findErrorNums(vector<int>& nums) {
        sort(nums.begin(),nums.end());
        vector<int> res(2);
        int n=nums.size();
        int r_s=n*(n+1)/2;
        int a_s=0;
        for(int num : nums) a_s+=num;
        for(int i=1;i<n;i++) {
            if(nums[i-1]==nums[i]) {
                res[0]=nums[i];
            }
        }
        res[1]=r_s-(a_s-res[0]);
        return res;
    }
};